src/NuGet.Clients/NuGet.MSSigning.Extensions/RepoSignCommand.cs

```diff
@@ -0,0 +1,114 @@
+       MinArgs = 1,
+       MaxArgs = 1,
+       UsageSummaryResourceName = "RepoSignCommandUsageSummary",
+       UsageExampleResourceName = "RepoSignCommandUsageExamples",
+       UsageDescriptionResourceName = "RepoSignCommandUsageDescription")]
+    public class RepoSignCommand : MSSignAbstract
+    {
+        // Default constructor used only for testing, since the Command Default Constructor is protected
+        public RepoSignCommand() : base()
+        {
+        }
+
+        private readonly List<string> _owners = new List<string>();
+
+        [Option(typeof(NuGetMSSignCommand), "RepoSignCommandOwnersDescription")]
+        public IList<string> Owners => _owners;
+
+        [Option(typeof(NuGetMSSignCommand), "RepoSignCommandV3ServiceIndexUrlDescription")]
+        public string V3ServiceIndexUrl { get; set; }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            using (var signRequest = GetRepositorySignRequest())
+            {
+                var packages = GetPackages();
+                var signCommandRunner = new SignCommandRunner();
+                var result = await signCommandRunner.ExecuteCommandAsync(
+                    packages, signRequest, Timestamper, Console, OutputDirectory, false, CancellationToken.None);
+
+                if (result != 0)
+                {
+                    throw new ExitCodeException(exitCode: result);
+                }
+            }
+        }
+
+        private RepositorySignPackageRequest GetRepositorySignRequest()
+        {
+            ValidatePackagePath();
+            WarnIfNoTimestamper(Console);
+            ValidateCertificateInputs();
+            EnsureOutputDirectory();
+            ValidatePackageOwners();
+
+            var signingSpec = SigningSpecifications.V1;
+            var signatureHashAlgorithm = ValidateAndParseHashAlgorithm(HashAlgorithm, nameof(HashAlgorithm), signingSpec);
+            var timestampHashAlgorithm = ValidateAndParseHashAlgorithm(TimestampHashAlgorithm, nameof(TimestampHashAlgorithm), signingSpec);
+            var certCollection = GetCertificateCollection();
+            var certificate = GetCertificate(certCollection);
+            var privateKey = GetPrivateKey(certificate);
+            var v3ServiceIndexUri = ValidateAndParseV3ServiceIndexUrl();
+
+            var request = new RepositorySignPackageRequest(
+                certificate,
+                signatureHashAlgorithm,
+                timestampHashAlgorithm,
+                v3ServiceIndexUri,
+                new ReadOnlyCollection<string>(Owners))
+            {
+                PrivateKey = privateKey
+            };
+
+            request.AdditionalCertificates.AddRange(certCollection);
+
+            return request;
+        }
+
+        private Uri ValidateAndParseV3ServiceIndexUrl()
+        {
+            // Assert mandatory argument
+            if (!string.IsNullOrEmpty(V3ServiceIndexUrl) ||
```

Comment:
I think it should be `&&` not `||`.

However, it seems simpler and equivalent to just write:
```
if (Uri.TryCreate(V3ServiceIndexUrl, UriKind.Absolute, out var v3ServiceIndexUrl))
{
    return v3ServiceIndexUrl;
}
```